File: JavaScript/Linked List/61_Rotate_List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
var rotateRight = function (head, k) {
    if (head === null || k === 0) return head;
    let temp = head, len = 1;
    while (temp.next !== null) {
        temp = temp.next;
        len++;
    }
    temp.next = head;
    let count = len - k % len;
    while (count > 0) {
        temp = temp.next;
        count--;
    }
    head = temp.next;
    temp.next = null;
    return head;
};

File: Java/Linked List/61_Rotate_List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || k == 0)
            return head;
        ListNode temp = head;
        int l = 1;
        while (temp.next != null) {
            temp = temp.next;
            l++;
        }
        temp.next = head;
        int count = l - k % l;
        while (count > 0) {
            temp = temp.next;
            count--;
        }
        head = temp.next;
        temp.next = null;
        return head;
    }
}